### DISCUSSION TITLE
How to Query Postgres Composite Types

### DISCUSSION THREAD

**Discussion by Aljabri-Salman** [supabase/supabase#2825] [Questions] [answered]
Let's say I have the following type and table:
```
CREATE TYPE inventory_item AS (
    name            text,
    supplier_id     integer,
    price           numeric
);

CREATE TABLE on_hand (
    item      inventory_item,
    count     integer
);
```
How to filter on_hand table based on item field in the client api. This is done in sql like:
```sql
select * from on_hand where (item).supplier_id  = 1
```
I tried doing this in flutter:
```dart
final response = await client
          .from('on_hand')
          .select()
          .filter("(item).supplier_id", "eq", 1)
          .execute();
```
I got this error:
> "failed to parser tree path ((item).supplier_id)" (line 1, column 1)

Is it possible to do this? if yes how?

**Comment by soedirgo** [comment]
@Aljabri-Salman this isn't possible right now, pending on https://github.com/PostgREST/postgrest/issues/1543. You can work around this by using a computed column as suggested in that issue, like so:
```sql
create or replace function on_hand_item_supplier_id(on_hand) returns integer as $$
  select $1.item.supplier_id;
$$ language sql;
```

And in flutter:

```dart
final response = await client
          .from('on_hand')
          .select()
          .eq("on_hand_item_supplier_id", 1)
          .execute();
```

**Comment by tanius** [comment, marked as answer]
This should be possible now, given that the upstream issue https://github.com/PostgREST/postgrest/issues/1543 has been resolved in the meantime. Syntax details are available in that issue's discussion. According to that, the syntax to use (in Flutter) would be:

```Dart
final response = await client
    .from('on_hand')
    .select()
    .filter("item->supplier_id", "eq", 1)
    .execute();
```

Or in general, the syntax has to result in this API call:

```HTTP
GET /on_hand?item->supplier_id=eq.1
```
